<?xml version="1.0"?>
<launch>
  <arg name="output" default="screen"/>
  <arg name="respawn" default="false"/>
  <arg name="start_manager" default="true"/>
  <arg name="start_server" default="false"/>
  <arg name="nodelet_manager" default="gazes_nodelet_manager"/>
  <arg name="launch-prefix" default=""/>

  <arg name="input" default="wp2/gaze"/>
  <arg name="global_frame_id" default="map"/>
  <arg name="nb_min_detection" default="10"/>
  <arg name="hfov" default="60.0"/>
  <arg name="aspect" default="1.3333"/>
  <arg name="output_world" default="gazes"/>

  <node name="$(arg nodelet_manager)"
        pkg="nodelet" type="nodelet"
        args="manager"
        respawn="$(arg respawn)"
        output="$(arg output)"
        launch-prefix="$(arg launch-prefix)"
        if="$(arg start_manager)"/>

  <node name="uwds_server"
        pkg="nodelet" type="nodelet"
        args="load uwds/UwdsServerNodelet $(arg nodelet_manager)"
        respawn="$(arg respawn)"
        output="$(arg output)"
        launch-prefix="$(arg launch-prefix)"
        if="$(arg start_server)"/>

  <node name="gazes_provider"
        pkg="nodelet" type="nodelet"
        args="load uwds_mummer_clients/GazesProvider $(arg nodelet_manager)"
        output="$(arg output)"
        launch-prefix="$(arg launch-prefix)">
    <remap from="~input" to="$(arg input)"/>
    <rosparam subst_value="true">
      global_frame_id: $(arg global_frame_id)
      hfov: $(arg hfov)
      aspect: $(arg aspect)
      nb_min_detection: $(arg nb_min_detection)
      output_world: $(arg output_world)
    </rosparam>
  </node>

</launch>
